### ISSUE TITLE
100% CPU Usage when using --listen-http

### ISSUE THREAD

**Issue by lukeescude** [sipwise/rtpengine#1520]
When rtpengine is started with --listen-http=localhost:3050, you can see 100% CPU usage on top.

Once that parameter is removed, rtpengine goes back to normal (which is very minimal CPU).

We have tested this and found the issue with 10.5, 10.4, and 10.3, running CentOS 7 x64

**Comment by lukeescude** [comment]
This appears to not be happening on 10.3.1.8, so a bug may have been introduced after this version.

**Comment by lukeescude** [comment]
I discovered that it will spike up to 100% just after issuing the command, and stay there for about 10 seconds, then go back down, so it’s not constantly at 100% apparently.

So, check top right after calling the jsonstats endpoint.

I will post the parameters I use to start rtpengine, maybe it’s some combination of factors.

**Comment by lukeescude** [comment]
Here is our unit file for rtpengine:

[Unit]
Description=rtpengine
Wants=network-online.target
After=network-online.target

[Service]
Type=simple
User=root
Group=root
LimitNOFILE=999999
Restart=always
RestartSec=5s
ExecStart=/usr/sbin/rtpengine --interface=(PUBLIC IP) --listen-ng=12221 --tos=184 --port-min=10000 --port-max=40000 timeout=30 --homer=(HOMER IP):9060 --homer-id=1 --homer-protocol=udp --log-level=3 --foreground

[Install]
WantedBy=multi-user.target

**Comment by rfuchs** [comment]
Yeah, still not reproducible. What version of libwebsockets do you have? My installation has `libwebsockets-3.0.1-2.el7.x86_64`

**Comment by lukeescude** [comment]
Same here:

`libwebsockets.x86_64               3.0.1-2.el7                     @epel`  
`libwebsockets-devel.x86_64         3.0.1-2.el7                     @epel`

Very strange, so it must be my environment somehow.

**Comment by rfuchs** [comment]
Are you using curl to make that request?

**Comment by lukeescude** [comment]
Interesting question, you've helped narrow it down actually.

Using curl and axios seems to be fine. However, it ramps up to 100% specifically when Zabbix (a monitoring system) queries `curl http://localhost:3050/cli/list/jsonstats` 

So it's something to do with the way Zabbix is polling it every 30 seconds (we have Zabbix collecting call quality metrics).

**Comment by rfuchs** [comment]
Use Wireshark or something to look at the HTTP request. It's probably some header that is different.

**Comment by lukeescude** [comment]
I was wrong, Axios and Zabbix both cause it. So cURL is the only safe way to query rtpengine it seems.

Yeah, I'll sniff it out and see what's being sent.

**Comment by lukeescude** [comment]
Some differences in the requests:

cURL: `Accept: */*`
Axios: `Accept: application/json, text/plain, */*`

Also, Axios also adds this header to the request: `Connection: close`

**Comment by lukeescude** [comment]
Yep this is how you can recreate the issue:

`curl http://localhost:3050/cli/list/jsonstats --header "Connection: close"`

It's that connection close header - Unfortunately, this is a very common header used for connection keepalives, so this may be unavoidable for a lot of use cases.

**Comment by rfuchs** [comment]
This does not happen with libwebsockets 2.0.3-3 from Debian Buster nor with libwebsockets 4.0.20-2 from Debian Bullseye. So perhaps something specific to the version used by CentOS.

**Comment by lukeescude** [comment]
Strange - Well we're altering the Connection header now on all requests, so the issue is "gone" so this can be closed if you think that is best.
